packages/cli/src/tools/windows/androidWinHelpers.ts

```diff
@@ -0,0 +1,301 @@
+import {readFile, writeFile} from 'fs';
+import {join} from 'path';
+import {executeCommand} from './executeWinCommand';
+import {getProcessorType} from '../processorType';
+import {promisify} from 'util';
+
+const readFileAsync = promisify(readFile);
+const writeFileAsync = promisify(writeFile);
+
+type HypervisorStatus = {
+  hypervisor: 'WHPX' | 'HAXM' | 'AMDH' | 'none';
+  installed: boolean;
+};
+
+/**
+ * Returns the path to where all Android related things should be installed
+ * locally to the user.
+ */
+export const getUserAndroidPath = () => {
+  return join(process.env.LOCALAPPDATA || '', 'Android');
+};
+
+/**
+ * Deals with ANDROID_HOME, ANDROID_SDK_ROOT or generates a new one
+ */
+export const getAndroidSdkRootInstallation = () => {
+  // TODO: Check if the path exists?
+  const env = process.env.ANDROID_SDK_ROOT || process.env.ANDROID_HOME;
+  const installPath = env
+    ? // Happens if previous installations or not fully completed
+      env
+    : // All Android zip files have a root folder, using `Android` as the common place
+      join(getUserAndroidPath(), 'Sdk');
+
+  return installPath;
+};
+
+/**
+ * Installs an Android component (e.g.: `platform-tools`, `emulator`)
+ * using the `sdkmanager` tool and automatically accepting the licenses.
+ */
+export const installComponent = (component: string, androidSdkRoot: string) => {
+  return new Promise((done, error) => {
+    const sdkmanager = join(androidSdkRoot, 'tools', 'bin', 'sdkmanager.bat');
+
+    const command = `"${sdkmanager}" --sdk_root="${androidSdkRoot}" "${component}"`;
+
+    const child = executeCommand(command);
+    let stderr = '';
+
+    child.stdout.on('data', data => {
+      if (data.includes('(y/N)')) {
+        child.stdin.write('y\n');
+      }
+    });
+
+    child.stderr.on('data', data => {
+      stderr += data.toString('utf-8');
+    });
+
+    child.on('close', data => {
```

Comment:
It would be better for readability to rename this parameter to something like `status` or even `exitStatus`.